### PR TITLE
Clean up docstyle in data, ml, and tune packages

### PR DESCRIPTION
**Pull request by ericl**
## Why are these changes needed?

Move these modules to fully following our style guide: https://docs.ray.io/en/latest/ray-contribute/getting-involved.html#code-style

Ideally this would be programmatically enforced (not sure the best way).

### CODE REVIEW

**Comment by bveeramani** [comment]
> Ideally this would be programmatically enforced (not sure the best way).

There's a tool called [pydocstyle](https://github.com/PyCQA/pydocstyle) by PyCQA, although I'm not sure if it can enforce no-type-in-docstring.

**Comment by ericl** [comment]
I've also come up with a pretty good regex, might try tackling this programatically in other modules when I have some spare time.

**Comment by ericl** [comment]
Ping for review.

**Review by richardliaw — APPROVED** [review]
lgtm, cc @xwjiang2010 @Yard1

**Review by Yard1 — APPROVED** [review]
lgtm to me too

File: python/ray/tune/automl/search_space.py [hunk 1]
```diff
@@ -14,7 +14,7 @@ def __init__(self, name):
         """Initialize ParameterSpace.
 
         Arguments:
-            name (str): Name of the parameter. Name can be dot separated,
+            name: Name of the parameter. Name can be dot separated,
```
Comment (krfricke) [inline python/ray/tune/automl/search_space.py:17]: In the cases where we remove the type annotation in the docstring, we should add it to the function definition instead, e.g. `def __init__(self, name: str):`

**Review by krfricke — DISMISSED** [review]
Thanks, generally LGTM, but we should move the type annotations to the function definition if possible, otherwise we'll lose information.

**Review by xwjiang2010 — APPROVED** [review]

**Review by scv119 — APPROVED** [review]

**Pull request merged**